packages/suite/src/actions/suite/trezorConnectActions.ts

```diff
@@ -1,154 +1,42 @@
-                });
-                return result;
-            };
-        });
-};
-
-export const init = () => async (dispatch: Dispatch, getState: GetState) => {
-    // set event listeners and dispatch as
-    TrezorConnect.on(DEVICE_EVENT, ({ event: _, ...action }) => {
-        // dispatch event as action
-        dispatch(action);
-    });
-
-    TrezorConnect.on(UI_EVENT, ({ event: _, ...action }) => {
-        // dispatch event as action
-        dispatch(action);
-    });
-
-    TrezorConnect.on(TRANSPORT_EVENT, ({ event: _, ...action }) => {
-        // dispatch event as action
-        dispatch(action);
-    });
-
-    TrezorConnect.on(BLOCKCHAIN_EVENT, ({ event: _, ...action }) => {
-        // dispatch event as action
-        dispatch(action);
-    });
-
-    const wrappedMethods = [
-        'getFeatures',
-        'getDeviceState',
-        'getAddress',
-        'ethereumGetAddress',
-        'rippleGetAddress',
-        'cardanoGetAddress',
-        'applySettings',
-        'changePin',
-        'pushTransaction',
-        'ethereumSignTransaction',
-        'signTransaction',
-        'rippleSignTransaction',
-        'cardanoSignTransaction',
-        'backupDevice',
-        'recoveryDevice',
-    ] as const;
-    wrappedMethods.forEach(key => {
-        // typescript complains about params and return type, need to be "any"
-        const original: any = TrezorConnect[key];
-        if (!original) return;
-        (TrezorConnect[key] as any) = async (params: any) => {
-            dispatch(lockDevice(true));
-            const result = await original(params);
-            dispatch(lockDevice(false));
-            return result;
-        };
-    });
-
-    dispatch(cardanoPatch());
-
-    try {
-        const connectSrc = resolveStaticPath('connect/');
-        // 'https://localhost:8088/';
-        // 'https://connect.corp.sldev.cz/develop/';
-
-        await TrezorConnect.init({
-            connectSrc,
-            transportReconnect: true,
-            debug: false,
-            popup: false,
-            webusb: isWeb(),
-            pendingTransportEvent: getState().devices.length < 1,
-            manifest: {
-                email: 'info@trezor.io',
-                appUrl: '@trezor/suite',
-            },
-        });
-
-        dispatch({
-            type: SUITE.CONNECT_INITIALIZED,
-        });
-    } catch (error) {
-        let formattedError: string;
-        if (typeof error === 'string') {
-            formattedError = error;
-        } else {
-            formattedError = error.code ? `${error.code}: ${error.message}` : error.message;
-        }
-        dispatch({
-            type: SUITE.ERROR,
-            error: formattedError,
-        });
+export const init = createAsyncThunk<
+    void,
+    void,
+    {
+        state: AppState;
     }
-};
+>('initConnect', (_, { dispatch, getState }) => {
+    const { enabledNetworks } = getState().wallet.settings;
```

Comment:
@Nodonisko ...or your version selectWalletSettings(getState())? What do you think?

Reply:
Yea I would prefer it selector way